<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ExtraSettingWindow</class>
 <widget class="QDialog" name="ExtraSettingWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>509</width>
    <height>863</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>其他设置</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>端口转发</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;在下面的输入框内输入端口转发规则，一行一条规则&lt;/p&gt;&lt;p&gt;例如，在内网主机 10.10.98.98 的 13389 端口上存在 RDP 远程桌面服务（TCP 协议），&lt;br/&gt;需要将其转发到本机 127.0.0.1:9850 使用，规则为：&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;TCP,127.0.0.1:9850,10.10.98.98:13389&lt;/span&gt;&lt;/p&gt;&lt;p&gt;支持的协议：TCP、UDP&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::TextInteractionFlag::LinksAccessibleByMouse|Qt::TextInteractionFlag::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="forwardTextEdit">
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="placeholderText">
         <string>TCP,127.0.0.1:9850,10.10.98.98:13389
UDP,127.0.0.1:25565,10.248.98.98:25565</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>自定义 DNS 解析（Hosts）</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;在下面的输入框内输入解析规则，一行一条规则&lt;/p&gt;&lt;p&gt;例如，将 google.com 解析为 127.0.0.1，规则为：&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;127.0.0.1 google.com&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::TextInteractionFlag::LinksAccessibleByMouse|Qt::TextInteractionFlag::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="hostsTextEdit">
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="placeholderText">
         <string>8.8.8.8 dns.google
127.0.0.1 www.mihoyo.com</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>强制 VPN 分流域名</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;在下面的输入框内输入强制 VPN 分流规则，一行一条规则&lt;/p&gt;&lt;p&gt;例如，要将 www.cnki.net 强制走 VPN 分流，规则为：&lt;br/&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;www.cnki.net&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::TextInteractionFlag::LinksAccessibleByMouse|Qt::TextInteractionFlag::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="proxyTextEdit">
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="placeholderText">
         <string>www.cnki.net</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>额外参数</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;在下面的输入框内输入额外的内核参数，可以用回车代替空格&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::TextInteractionFlag::LinksAccessibleByMouse|Qt::TextInteractionFlag::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="extraArgTextEdit">
        <property name="font">
         <font>
          <family>Consolas</family>
         </font>
        </property>
        <property name="placeholderText">
         <string>--version</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ExtraSettingWindow</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ExtraSettingWindow</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
